wurst/systems/spells/SmokeMeat.wurst

```diff
@@ -21,11 +21,21 @@ init
                     case ITEM_DISEASED_MEAT
                         charges += floor(maybeMeat.getCharges() / 2)
 
+            //Spawn meat in appropriately sized stacks
             if charges > 0
                 removeMeats(smoker)
-                let meat = createItem(ITEM_SMOKED_MEAT, smoker.getPos())
-                ..setCharges(charges)
-                smoker.stackItem(meat)
+                let maxStack = getItemMaxStacks(ITEM_SMOKED_MEAT)
+                while charges > 0
+                    if charges > maxStack
+                        let meat = createItem(ITEM_SMOKED_MEAT, smoker.getPos())
```

Comment:
You can condense this without an `if`.

Use `..setCharges(min(charges, maxStack))` and `charges = max(0, charges - maxStack)`.

I'd rename `charges` to `remainingCharges` to be clearer.